src/service/resources-impl.js

```diff
@@ -387,15 +388,40 @@ export class Resources {
     this.resources_.push(resource);
 
     if (this.isRuntimeOn_) {
-      // Try to immediately build element, it may already be ready.
-      resource.build(this.forceBuild_);
-      this.schedulePass();
+      if (this.documentReady_) {
+        // Try to immediately build element, it may already be ready.
+        resource.build();
+        this.schedulePass();
+      } else {
+        this.pendingBuildResources_.push(element);
+        this.buildReadyResources_();
+      }
     }
 
     dev.fine(TAG_, 'element added:', resource.debugid);
   }
 
   /**
+   * Builds resources that are ready to be built.
+   * @private
+   */
+  buildReadyResources_() {
```

Comment:
Debouncing might have a negative effect here. Since elements are probably added pretty fast, debouncing might just delay our execution till the whole document was parsed - this would defeats the purpose. I am not sure if there's an easy way to test this out, but maybe a throttling approach would work better where it won't delay the execution but instead limits how frequent this is being called.

@dvoytenko any idea if this is a concern or is it safe to just call it on every element add?